pkg/deploy/deploy_test.go

```diff
@@ -1379,33 +1412,105 @@ func TestDeployConfigFF(t *testing.T) {
 	}
 }
 
-func TestDeployDryRun(t *testing.T) {
-	c := dynatrace.EnvironmentClients{
-		dynatrace.EnvironmentInfo{Name: "env", Group: "group"}: &client.DummyClientSet,
-	}
+type StubOpenPipelineClient struct {
+	UpdateStub func() (openpipeline.Response, error)
+}
+
+func (c *StubOpenPipelineClient) GetAll(ctx context.Context) ([]openpipeline.Response, error) {
+	return []coreapi.Response{}, nil
+}
+
+func (c *StubOpenPipelineClient) Update(ctx context.Context, id string, data []byte) (openpipeline.Response, error) {
+	return c.UpdateStub()
+}
+
+func TestLogResponseErrors(t *testing.T) {
 	projects := []project.Project{
 		{
 			Configs: project.ConfigsPerTypePerEnvironments{
 				"env": project.ConfigsPerType{
 					"p1": {
 						config.Config{
-							Type:        config.Segment{},
+							Type:        config.OpenPipelineType{},
 							Environment: "env",
 							Coordinate: coordinate.Coordinate{
 								Project:  "p1",
-								Type:     "segment",
+								Type:     "type",
 								ConfigId: "config1",
 							},
-							Template: testutils.GenerateDummyTemplate(t),
 						},
 					},
 				},
 			},
 		},
+	}
+
+	tests := []struct {
+		name           string
+		updateStub     func() (openpipeline.Response, error)
+		expectedErrLog string
+	}{
+		{
+			name: "500 error",
+			updateStub: func() (openpipeline.Response, error) {
+				return openpipeline.Response{}, coreapi.APIError{StatusCode: 501}
+			},
+			expectedErrLog: "Deployment failed - Dynatrace Server Error",
+		},
+		{
+			name: "400 error",
+			updateStub: func() (openpipeline.Response, error) {
+				return openpipeline.Response{}, coreapi.APIError{StatusCode: 404}
+			},
+			expectedErrLog: "Deployment failed - Dynatrace API rejected HTTP request / JSON data",
+		},
+		{
+			name: "default error",
+			updateStub: func() (openpipeline.Response, error) {
+				return openpipeline.Response{}, coreapi.APIError{StatusCode: 0}
+			},
+			expectedErrLog: "Deployment failed - Dynatrace API call unsuccessful",
+		},
+	}
+	logSpy := bytes.Buffer{}
+	log.SetDefaultLogger(loggers.LogOptions{JSONLogging: true, LogSpy: &logSpy})
```

Comment:
What was again the reason, why `log.PrepareLogging` was not used? Too many parameters tests don't need to care about?

Reply:
That is a very good point. To be honest, I just didn't see it :). I'll have a closer look.